Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabComponentAdapter.cpp

```diff
@@ -32,9 +38,23 @@ namespace AzToolsFramework::Prefab
 
     void PrefabComponentAdapter::SetComponent(AZ::Component* componentInstance)
     {
+        auto owningInstance = AZ::Interface<InstanceEntityMapperInterface>::Get()->FindOwningInstance(
+            componentInstance->GetEntityId());
+        AZ_Assert(owningInstance.has_value(), "Entity owning the component doesn't have an owning prefab instance.");
+        auto entityAlias = owningInstance->get().GetEntityAlias(componentInstance->GetEntityId());
+        AZ_Assert(entityAlias.has_value(), "Owning entity of component doesn't have a valid entity alias in the owning prefab.");
+        m_entityAlias = entityAlias->get();
+
         // Set the component alias before calling SetValue() in base SetComponent().
         // Otherwise, an empty component alias will be used in DOM data.
         m_componentAlias = componentInstance->GetSerializedIdentifier();
+
+        if (m_componentAlias.empty())
+        {
+            AZStd::string componentAlias(AZStd::string::format("Component_[%llu]", componentInstance->GetId()));
+            componentInstance->SetSerializedIdentifier(componentAlias);
+            m_componentAlias = AZStd::move(componentAlias);
+        }
```

Comment:
A reasonable workaround for fixing the empty alias issue😊

Looks like it can be simplified to this:
```cpp
m_componentAlias = AZStd::string::format("Component_[%llu]", componentInstance->GetId());
componentInstance->SetSerializedIdentifier(m_componentAlias);
```

Reply:
Actually, this should be removed. I was just testing something and added this. Since @michabr is working on this problem, I'll let her fix the empty alias problem